tests/Dialect/Polynomial/Conversions/heir_polynomial_to_llvm/runner/lower_ntt_perf_runner.mlir

```diff
@@ -1,4 +1,5 @@
-// RUN: heir-opt %s --heir-polynomial-to-llvm \
+// RUN: heir-opt %s --polynomial-to-mod-arith \
+// RUN:  --mod-arith-to-arith --heir-polynomial-to-llvm \
```

Comment:
I cloned your PR to my local machine and I am able to run `heir-polynomial-to-llvm` on this file without the extra pass flags. What error are you seeing that makes you think you should add this?

To be more specific, I ran

```
bazel run --noallow_analysis_cache_discard //tools:heir-opt -- --heir-polynomial-to-llvm $PWD/tests/Dialect/Polynomial/Conversions/heir_polynomial_to_llvm/runner/lower_ntt_perf_runner.mlir
```

It may help to add the flags `--mlir-print-ir-after-all --mlir-print-ir-tree-dir=/tmp/mlir` and that will print the IR after each pass to a file so you can inspect it more easily than `--debug`.

Reply:
FWIW, I also removed the flags and the tests still pass.

Reply:
Just to confirm that you only mean this specific testcase?

I was able to remove the flags on this test but not any of the others locally. I had just been a bit overzealous when adding the flags, so I don't have any failures locally for this testcase either.

Reply:
In principle this shouldn't be a problem because those passes are in the broader pipeline. We can submit as-is, but please put a TODO where it is strictly necessary (and if possible, post the error you see in the issue).

Reply:
Narrowing down on a fix to the `ElementwiseToAffine` pass, need to step away for now but will update the pr later today.